ZAPD/Globals.cpp

```diff
@@ -82,47 +98,67 @@ void Globals::ReadConfigFile(const std::string& configFilePath)
 	if (root == nullptr)
 		return;
 
-	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != NULL;
+	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != nullptr;
 	     child = child->NextSiblingElement())
 	{
-		if (std::string(child->Name()) == "SymbolMap")
-		{
-			std::string fileName = std::string(child->Attribute("File"));
-			GenSymbolMap(Path::GetDirectoryName(configFilePath) + "/" + fileName);
-		}
-		else if (std::string(child->Name()) == "Segment")
+		auto it = ConfigTypeDictionary.find(child->Name());
+		if (it != ConfigTypeDictionary.end())
```

Comment:
IMO it would be better to put the failure branch first and `continue`, so as to reduce the indentation level for the rest of the code.

Reply:
Oh, this looks much better, thanks. Is there a general rule about if error cases go first or last?

Reply:
I don't have an answer for ZAPD (you'd have to ask @NEstelami) but early returns/exits are common in many other projects (e.g. [LLVM](https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code))

Reply:
Yeah, decomp projects are an outlier here in enforcing else/ifs everywhere. I think makes the flow much more complicated and creates so much indenting, I wish we could switch over to using returns.